var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -55,9 +65,14 @@ class Lammps(CMakePackage, CudaPackage):
     version('20170901', sha256='5d88d4e92f4e0bb57c8ab30e0d20de556830af820223778b9967bec2184efd46')
 
     def url_for_version(self, version):
-        vdate = dt.datetime.strptime(str(version), "%Y%m%d")
-        return "https://github.com/lammps/lammps/archive/patch_{0}.tar.gz".format(
-            vdate.strftime("%d%b%Y").lstrip('0'))
+        split_ver = str(version).split('.')
+        vdate = dt.datetime.strptime(split_ver[0], "%Y%m%d")
+        if len(split_ver) < 2:
```

Comment:
@adamjstewart if you have python suggestions here, go ahead.

Reply:
My only suggestion would be to use versions like 2021-10-27 instead of 20211027 as it's easier to read and easier to say something like `depends_on('lammps@2021:')`. Not sure if that version number change would break other packages/users.

Reply:
I don't want to change the version as `YYYYMMDD` is lammps internal cmake version.